addons/resource/models/resource.py

```diff
@@ -343,7 +346,7 @@ def _iter_day_attendance_intervals(self, day_date, start_time, end_time):
             yield self._interval_new(dt_f, dt_t, {'attendances': calendar_working_day})
 
     @api.multi
-    def _get_day_work_intervals(self, day_date, start_time=None, end_time=None, compute_leaves=False, resource_id=None):
+    def _get_day_work_intervals(self, day_date, start_time=None, end_time=None, compute_leaves=False, resource_id=None, time_type='leave'):
```

Comment:
I would rather add a leave_domain additional parameter (leave_domain=None). If leave_domain is None, its default value would actually be [('time_type', '=', 'leave')]; other other value than None would be passed.

That way we have the default computation for time_type while keeping the possibiltiy to customize compeltely the way we compute leave intervals.